lib/active_merchant/billing/gateways/quickbooks.rb

```diff
@@ -170,31 +192,38 @@ def commit(uri, body = {}, method = :post)
         # The QuickBooks API returns HTTP 4xx on failed transactions, which causes a
         # ResponseError raise, so we have to inspect the response and discern between
         # a legitimate HTTP error and an actual gateway transactional error.
+        headers = {}
         response =
           begin
             case method
             when :post
-              ssl_post(endpoint, post_data(body), headers(:post, endpoint))
+              headers = headers(:post, endpoint)
```

Comment:
When squinting at this logic, the get and post logic do almost the same thing. Is there a little bit of clean up that could be done here to get rid of the conditional for get and post?

Reply:
updated!